components/eam/src/control/apply_iop_forcing.F90

```diff
@@ -0,0 +1,275 @@
+    do m=1,pcnst
+      q_update(k,m) = q_in(k,m) + scm_dt*q_lsf(k,m)
+    end do
+  enddo
+
+  !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+  !  Set U and V fields
+
+  if (have_v .and. have_u .and. .not. dp_crm) then
+    do k=1,plev
+      u_update(k) = uobs(k)
+      v_update(k) = vobs(k)
+    enddo
+  else
+    do k=1,plev
+      u_update(k) = u_in(k)
+      v_update(k) = v_in(k)
+    enddo
+  endif
+
+end subroutine advance_iop_forcing
+
+!=========================================================================
+
+subroutine advance_iop_nudging(scm_dt, ps_in, t_in, q_in, &       ! In
+                             t_update, q_update, relaxt, relaxq ) ! Out
+
+!-----------------------------------------------------------------------
+!
+! Purpose:
+! Option to nudge t and q to observations as specified by the IOP file
+!-----------------------------------------------------------------------
+
+  ! Input arguments
+  real(r8), intent(in) :: scm_dt           ! model time step [s]
+  real(r8), intent(in) :: ps_in            ! surface pressure [Pa]
+  real(r8), intent(in) :: t_in(plev)       ! temperature [K]
+  real(r8), intent(in) :: q_in(plev)       ! water vapor mixing ratio [kg/kg]
+
+  ! Output arguments
+  real(r8), intent(out) :: t_update(plev)  ! updated temperature [K]
+  real(r8), intent(out) :: q_update(plev)  ! updated water vapor [kg/kg]
+  real(r8), intent(out) :: relaxt(plev)    ! relaxation of temperature [K/s]
+  real(r8), intent(out) :: relaxq(plev)    ! relaxation of vapor [kg/kg/s]
+
+  ! Local variables
+  integer :: k, nlon
+  real(r8) rtau(plev)
+  real(r8) pmidm1(plev)  ! pressure at model levels
+  real(r8) pintm1(plevp) ! pressure at model interfaces
+  real(r8) pdelm1(plev)  ! pdel(k)   = pint  (k+1)-pint  (k)
+
+  nlon = 1 ! number of columns for plevs0 routine
+  call plevs0(nlon, plon, plev, ps_in, pintm1, pmidm1, pdelm1)
+
+  ! Set relaxation arrays to zero
+  do k=1,plev
+    relaxt(k) = 0.0_r8
+    relaxq(k) = 0.0_r8
+  end do
+
+  do k=1,plev
+
+    if (pmidm1(k) .le. iop_nudge_tq_low*100._r8 .and. &
+      pmidm1(k) .ge. iop_nudge_tq_high*100._r8) then
+
+      ! Set the relaxation time scale
+      rtau(k)   = iop_nudge_tscale
+      rtau(k)   = max(scm_dt,rtau(k))
+      relaxt(k) = -(t_update(k) - tobs(k))/rtau(k)
+      relaxq(k) = -(q_update(k) - qobs(k))/rtau(k)
+
+      t_update(k) = t_update(k) + relaxt(k)*scm_dt
+      q_update(k) = q_update(k) + relaxq(k)*scm_dt
+
+    endif
+
+  end do
+
+end subroutine advance_iop_nudging
+
+!=========================================================================
+
+subroutine advance_iop_subsidence(scm_dt, ps_in, &                    ! In
+                             u_in, v_in, t_in, q_in, &                ! In
+                             u_update, v_update, t_update, q_update ) ! Out
+
+!-----------------------------------------------------------------------
+!
+! Purpose:
+! Option to nudge t and q to observations as specified by the IOP file
```

Comment:
This comment seems to be copied from the routine above, so it would be good to update it.

Reply:
Good find.  This has been updated.